colcon_bundle/verb/_overlay_utilities.py

```diff
@@ -135,14 +158,14 @@ def recursive_tar_gz_in_path(output_path, path):
             tar.add(some_path, arcname=os.path.basename(some_path))
 
 
-def _generate_template(template_name, script_name, context_vars):
+def _generate_template(template_name, script_dest, context_vars):
```

Comment:
This operation is really called "rendering", not "generating", and let's add type annotations while we're fixing this up.

This package is Python3, right? If so, can we please update to use `pathlib.Path` across what you're touching here? It makes it a lot easier to reason about paths

```suggestion
def _render_template(template_name: Path, script_dest: Path, context_vars: dict):
```